src/backend/core/api/viewsets.py

```diff
@@ -191,6 +198,142 @@ def get_me(self, request):
         )
 
 
+class StatsViewSet(viewsets.GenericViewSet):
+    """
+    API endpoint that provides various statistics in JSON format.
+    """
+
+    permission_classes = [IsAdminUser]
+
+    def list(self, request):
+        user_timezone = "UTC"
+        if request.user.is_authenticated:
+            user_timezone = str(getattr(request.user, "timezone", user_timezone))
+
+        try:
+            timezone_to_use = ZoneInfo(user_timezone)
+        except ZoneInfoNotFoundError:
+            return drf_response.Response(
+                {"detail": "Invalid user timezone provided."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        if "tz" in request.query_params:
+            try:
+                timezone_to_use = ZoneInfo(request.query_params["tz"])
+            except ZoneInfoNotFoundError:
+                return drf_response.Response(
+                    {"detail": "Invalid timezone provided."},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+        now_time = now()
+        now_in_user_tz = now_time.astimezone(timezone_to_use)
+        today = now_in_user_tz.replace(hour=0, minute=0, second=0, microsecond=0)
+        one_week_ago = today - timedelta(days=7)
+        one_month_ago = today - timedelta(days=30)
```

Comment:
See previous comment + this is quite false because of daylight saving time.

``` python
one_week_ago = datetime.datetime.combine(today - timedelta(days=7), datetime.time.min)
thirty_days_ago = datetime.datetime.combine(today - timedelta(days=30), datetime.time.min)
```